src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -444,6 +444,7 @@ public void stopJenkins() throws Throwable {
 
     public void runRemotely(Step s) throws Throwable {
         HttpURLConnection conn = (HttpURLConnection) endpoint("step").openConnection();
+        conn.setRequestProperty("Content-Type", "text/plain; utf-8");
```

Comment:
What? It is binary, not text.

Reply:
It is binary but when a filter does a request.getParameter it invalidates the data hence an error when reading the data stream. 
If it's send with that Content Type, request.getParam goes well and the getInputStream works smoothly.

Reply:
Weird. Definitely deserves to be noted in a comment here that the content type is totally incorrect.

Reply:
Indeed. After this comment, I tried out another Content-Type (what if missing the  content-type was the problem).
And if worked with application/octec-stream, which makes more sense that text.

Reply:
So IIUC merely adding _any_ `Content-Type` header fixes the problem?